### PR TITLE
NEW: test edit Topic properties

### PR DESCRIPTION
**Pull request by asanzgom**
Closes [MGDSTRM-10370](https://issues.redhat.com/browse/MGDSTRM-10370)

### CODE REVIEW

File: tests/kas_with_instance.spec.ts [hunk 1]
```diff
@@ -183,3 +183,75 @@ test('create and delete a Kafka Topic', async ({ page }) => {
...
+  await row.locator('[aria-label="Actions"]').click();
+  await page.getByText('Edit topic configuration').click();
+
+  // store previous num partitions value to compare afterwards
+  const numPartitionsBefore: string = await page.locator('input[name="num-partitions"]').getAttribute('value');
+
+  for (let i = 0; i < 2; i++) {
+    await page.locator('button[name="num-partitions"]').nth(1).click();
+  }
+  await page.locator('button[name="num-partitions"]').first().click();
```
Comment (Frawless) [inline tests/kas_with_instance.spec.ts:204]: I am not sure if I follow the steps. What exactly does this do?

File: tests/kas_with_instance.spec.ts [hunk 2]
```diff
@@ -183,3 +183,75 @@ test('create and delete a Kafka Topic', async ({ page }) => {
...
+
+  for (let i = 0; i < 2; i++) {
+    await page.locator('button[name="retention-ms"]').nth(1).click();
+  }
+  await page.locator('button[name="retention-ms"]').first().click();
+
+  await page.locator('button:has-text("days")').click();
+  await page.getByText('hours').click();
```
Comment (Frawless) [inline tests/kas_with_instance.spec.ts:212]: This looks to me like a very vague selector.

Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:212]: there is a problem with the units select as it is not a select, but a button (the outcome from Patternfly)

File: tests/kas_with_instance.spec.ts (hunk 1 above)
Comment (Frawless) [reply tests/kas_with_instance.spec.ts:204]: so you click on the small buttons available in the field, increase it to 2 and then decrease it? If yes, this deserves a comment

File: tests/kas_with_instance.spec.ts (hunk 2 above)
Comment (Frawless) [reply tests/kas_with_instance.spec.ts:212]: so `button:hast-text("hours")`?

File: tests/kas_with_instance.spec.ts (hunk 1 above)
Comment (Frawless) [reply tests/kas_with_instance.spec.ts:204]: Also, you should probably check the value

File: tests/kas_with_instance.spec.ts (hunk 2 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:212]: yes, there is no `select` to locate for: 
image

File: tests/kas_with_instance.spec.ts (hunk 1 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:204]: values are checked at the bottom of the test

image

File: tests/kas_with_instance.spec.ts (hunk 2 above)
Comment (Frawless) [reply tests/kas_with_instance.spec.ts:212]: I think there is a way (not sure if this works for my suggestion) to pick a specific element with text somewhere inside so button with hours (`button >> has-text(hours)` maybe?) should work

File: tests/kas_with_instance.spec.ts (hunk 1 above)
Comment (Frawless) [reply tests/kas_with_instance.spec.ts:204]: but you increase the value (line 202) and right after decreasing it (line 204), you should check the value between. Or am I missing something?

Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:204]: in the end it is just changing the number from default value, but on the same test we make sure the + and - are working. We increase by 2 and decrease by 1

File: tests/kas_with_instance.spec.ts (hunk 2 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:212]: I have tried some workarounds but in the end I just stuck with the PW debugger outcome at recording which is assured to work

File: tests/kas_with_instance.spec.ts (hunk 1 above)
Comment (Frawless) [reply tests/kas_with_instance.spec.ts:204]: ok, I guess I can live with it. You can reuse the code via function as it is same as bellow. Also wouldn't be better to use `nth(0)` instead of `.first()` ?

File: tests/kas_with_instance.spec.ts (hunk 2 above)
Comment (Frawless) [reply tests/kas_with_instance.spec.ts:212]: As long as it will work and some else will approve (with more experience than I have), I can live with it. However, I think the selector could be better. Feel free to resolve this.

Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:212]: Replaced for `await page.locator('button', { hasText: 'hours' }).click();`

File: tests/kas_with_instance.spec.ts (hunk 1 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts]: Replaced for `await numPartitionsButton.nth(0).click();`

**Comment by asanzgom** [comment]
Jenkins retest this

**Review by kornys — APPROVED** [review]

File: tests/kas_with_instance.spec.ts [hunk 3]
```diff
@@ -183,3 +183,89 @@ test('create and delete a Kafka Topic', async ({ page }) => {
...
+  await navigateToKafkaTopicsList(page, testInstanceName);
+  await createKafkaTopic(page, testTopicName);
+
+  const row = page.locator('tr', { hasText: testTopicName });
+  await row.locator('[aria-label="Actions"]').click();
+  await page.getByText('Edit topic configuration').click();
+
+  await expect(page.locator('input[name="num-partitions"]')).toHaveValue('1', { timeout: 3000 });
+  const numPartitionsBefore: string = await page.locator('input[name="num-partitions"]').getAttribute('value');
+  console.log('numPartitionsBefore: ' + numPartitionsBefore);
```
Comment (Frawless) [inline tests/kas_with_instance.spec.ts:198]: maybe you could use a more meaning full message?

File: tests/kas_with_instance.spec.ts [hunk 4]
```diff
@@ -183,3 +183,89 @@ test('create and delete a Kafka Topic', async ({ page }) => {
...
+// test_4kafka.py test_edit_topic_properties_after_creation
+test('edit topic properties after creation', async ({ page }) => {
+  await navigateToKafkaTopicsList(page, testInstanceName);
+  await createKafkaTopic(page, testTopicName);
+
+  const row = page.locator('tr', { hasText: testTopicName });
+  await row.locator('[aria-label="Actions"]').click();
+  await page.getByText('Edit topic configuration').click();
+
+  await expect(page.locator('input[name="num-partitions"]')).toHaveValue('1', { timeout: 3000 });
```
Comment (Frawless) [inline tests/kas_with_instance.spec.ts:196]: could you please add a comment why is this needed?

File: tests/kas_with_instance.spec.ts [hunk 5]
```diff
@@ -183,3 +183,89 @@ test('create and delete a Kafka Topic', async ({ page }) => {
...
+  const numPartitionsBefore: string = await page.locator('input[name="num-partitions"]').getAttribute('value');
+  console.log('numPartitionsBefore: ' + numPartitionsBefore);
+  const numPartitionsButton = page.locator('button[name="num-partitions"]');
+  for (let i = 0; i < 2; i++) {
+    await numPartitionsButton.nth(1).click();
+  }
+  await numPartitionsButton.nth(0).click();
+  await expect(page.locator('input[name="num-partitions"]')).not.toHaveValue(numPartitionsBefore);
```
Comment (Frawless) [inline tests/kas_with_instance.spec.ts:204]: Some comment would be nice as well

File: tests/kas_with_instance.spec.ts [hunk 6]
```diff
@@ -183,3 +183,89 @@ test('create and delete a Kafka Topic', async ({ page }) => {
...
+
+  // Retention Time
+  await expect(page.locator('label:has-text("days") input[type="number"]')).toHaveCount(1);
+  page.locator('label:has-text("days") input[type="number"]').click();
+  const retentionTimeBefore = await page.locator('label:has-text("days") input[type="number"]').getAttribute('value');
+  const retentionTimeButton = page.locator('button[name="retention-ms"]');
+  for (let i = 0; i < 2; i++) {
+    await retentionTimeButton.nth(1).click();
+  }
+  await retentionTimeButton.nth(0).click();
```
Comment (Frawless) [inline tests/kas_with_instance.spec.ts:214]: Same as above

File: tests/kas_with_instance.spec.ts [hunk 7]
```diff
@@ -183,3 +183,89 @@ test('create and delete a Kafka Topic', async ({ page }) => {
...
+
+  // Retention Size
+  await expect(page.locator('label:has-text("bytes") input[type="number"]')).toHaveCount(1);
+  await page.locator('label:has-text("bytes") input[type="number"]').click();
+  const retentionSizeBefore = await page.locator('label:has-text("bytes") input[type="number"]').getAttribute('value');
+  const retentionSizeButton = page.locator('button[name="retention-bytes"]');
+  for (let i = 0; i < 2; i++) {
+    await retentionSizeButton.nth(1).click();
+  }
+  await retentionSizeButton.nth(0).click();
```
Comment (Frawless) [inline tests/kas_with_instance.spec.ts:228]: Same as above

File: tests/kas_with_instance.spec.ts (hunk 3 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:198]: what do you suggest @Frawless ?

File: tests/kas_with_instance.spec.ts (hunk 4 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:196]: sure

File: tests/kas_with_instance.spec.ts (hunk 5 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:204]: added

File: tests/kas_with_instance.spec.ts (hunk 6 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:214]: added

File: tests/kas_with_instance.spec.ts (hunk 3 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:198]: added `console.log('Number of partitions by default: ' + numPartitionsBefore);`

File: tests/kas_with_instance.spec.ts (hunk 6 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:217]: Can I resolve this conversation @Frawless ?

File: tests/kas_with_instance.spec.ts (hunk 7 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:232]: Can I resolve this conversation @Frawless ?

File: tests/kas_with_instance.spec.ts (hunk 4 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts:197]: Can I resolve this conversation @Frawless ?

File: tests/kas_with_instance.spec.ts (hunk 3 above)
Comment (asanzgom) [reply tests/kas_with_instance.spec.ts]: Can I resolve this conversation @Frawless ?

**Review by Frawless — APPROVED** [review]
